contracts/extensions/collections/HasSlashIndicatorContract.sol

```diff
@@ -9,22 +9,39 @@ contract HasSlashIndicatorContract is IHasSlashIndicatorContract, HasProxyAdmin
   ISlashIndicator internal _slashIndicatorContract;
 
   modifier onlySlashIndicatorContract() {
-    if (slashIndicatorContract() != msg.sender) revert ErrCallerMustBeSlashIndicatorContract();
+    _requiresSlashIndicatorContract();
```

Comment:
Do we need a internal method here?

Reply:
compiler will inline modifier code to the function, therefore extracting the logic to internal fn will save code size since it will jump into the fn instead of duplicating the logic inside modifier to many places.
reference: https://youtu.be/XDqD3X8DCiw